src/manifest.js

```diff
@@ -0,0 +1,22 @@
+const { resolve } = require('path')
+const { arch, platform } = require('process')
+
+const { safeWriteFile } = require('./utils/fs')
+
+const MANIFEST_VERSION = 1
+
+const createManifest = async ({ functions, path }) => {
+  const formattedFunctions = functions.map(formatFunction)
+  const payload = {
+    functions: formattedFunctions,
+    system: { arch, platform },
+    timestamp: Date.now(),
+    version: MANIFEST_VERSION,
+  }
+
+  await safeWriteFile(path, JSON.stringify(payload))
```

Comment:
Should exceptions be propagated instead?

Reply:
I'm glad you brought this up, because I was on the fence about it.

I did it this way because the existence of the manifest doesn't impact the correctness of the bundle so I thought the operation should still succeed. However, since the manifest creation is opt-in, it makes more sense for any errors to be propagated, as you say.

Done in https://github.com/netlify/zip-it-and-ship-it/pull/601/commits/3c53e0f124d58ee673847a63a28808d9e803090f.